modules/service/src/main/scala/lucuma/odb/graphql/input/TimingWindowEndInput.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package input
+
+import cats.syntax.all.*
+import edu.gemini.grackle.Result
+import lucuma.core.util.TimeSpan
+import lucuma.core.util.Timestamp
+import lucuma.odb.graphql.binding.*
+
+case class TimingWindowEndInput(
+  atUtc: Option[Timestamp],
+  after: Option[TimeSpan],
+  repeat: Option[TimingWindowRepeatInput]
+)
+
+object TimingWindowEndInput:
+  object messages:
+    val OnlyOneDefinition: String = "Only one of atUtc or after may be specified."
+    val RepeatOnlyWhenAfter: String = "repeat can only be specified when after is specified."
+    val RepeatPeriodGreaterThanAfter: String = "repeat.period must be greater than after."
```

Comment:
Do you want the dot between `repeat` and `period` in the string?

Reply:
The idea is to indicate that this value is invalid:
```
timingWindows: [{
  ...
  repeat: {
    period: XXX
  }
}]
```

Are we using another notation for that?

Reply:
> Are we using another notation for that?

I don't know. This is fine with me.